docs/guides/basic-rbac.mdx

```diff
@@ -0,0 +1,282 @@
+
+## Create an admin dashboard and protect it
+
+In your application, create a file named `page.tsx` in the `src/app/admin/dashboard` directory. You will likely need to create both the `admin` and `dashboard` directories when doing this. Copy the following code and paste it into the file.
+
+```tsx filename="src/app/admin/dashboard/page.tsx
+import { auth } from "@clerk/nextjs";
+import { redirect } from "next/navigation";
+
+export default function AdminDashboard() {
+  const { sessionClaims } = auth();
+
+  if (sessionClaims?.metadata.role !== "admin") {
+    return redirect("/");
+  }
+
+  return (
+    <>
+      <h1>This is the admin dashboard</h1>
+      <p>This page is restricted to users with the 'admin' role</p>
+    </>
+  );
+}
+```
+
+The `/admin/dashboard` route now requires that the user is logged into the application. It also requires that the user have a publicMetadata of `{"role": "admin" }`
+
+## Create a reusable function
+
+Let's create a helper function to make checking roles easier. The first step is modifying `globals.d.ts`. Create a `type` for `Roles` so that union type for the roles can be used in other places in the application, then modify the interface we previously added to use the new `Roles` type.
+
+```ts filename="src/types/globals.d.ts {3, 8}
+export { };
+
+export type Roles = "admin" | "moderator"
+
+declare global {
+  interface CustomJwtSessionClaims {
+    metadata: {
+      role?: Roles
+    };
+  }
+}
+```
+
+The next step is creating the helper function. Create the `src/utils` directory and inside add the file 'roles.ts'. Add the following code to that file. This function will accept a role using that `Roles` type, and then return `true` if the user has that role and `false` if the user does not.
+
+```ts filename="src/utils/roles.ts 
+import { Roles } from "@/types/globals"
+import { auth } from "@clerk/nextjs"
+
+export const checkRole = (role: Roles) => {
+  const { sessionClaims } = auth()
+  return sessionClaims?.metadata.role !== role
+}
+```
+
+The last step to implement the helper function is refactoring the admin dashboard to use it. In the `if()` statement, remove the check that was used previously and repalce it with the new `checkRole()` function and pass that function `"admin"`. 
+
+```tsx filename="src/app/admin/dashboard/page.tsx {1, 6-8}
+import { redirect } from "next/navigation";
+import { checkRole } from "@/utils/roles";
+
+export default function AdminDashboard() {
+
+  if (checkRole("admin")) {
+    return redirect("/");
+  }
+
+  return (
+    <>
+      <h1>This is the admin dashboard</h1>
+      <p>This page is restricted to users with the 'admin' role</p>
+    </>
+  );
+}
+```
+
+<Callout type="info">
+You can modify the behaviour of the helper function to meet your needs. Maybe it will return the roles that the user has, or you could create a `protectByRole()` and have that function handle the redirect.
+</Callout>
+
+## Add admin tools to find users and add roles
```

Comment:
Consider adding a brief explanation of the useFormState hook and how it is used in the ManageRoles, in the summary at the top here on 145 & 147.

Reply:
Good catch. I'll get that added shortly.